astropy/cosmology/io/table.py

```diff
@@ -11,20 +11,20 @@
 from .utils import convert_parameter_to_column
 
 
-def from_table(table, index=None, *, move_to_meta=False, cosmology=None):
+def from_table(table, index=None, *, move_to_meta=False, cosmology=None, rename=None):
     """Instantiate a `~astropy.cosmology.Cosmology` from a |QTable|.
 
     Parameters
     ----------
     table : `~astropy.table.Table`
         The object to parse into a |Cosmology|.
-    index : int, str, or None, optional
+    index : int or str or None, optional
```

Comment:
Not sure about changes like these - unrelated to the PR and not an improvement since grammatically incorrect. Is this to help computers or humans?

Reply:
This comes from https://numpydoc.readthedocs.io/en/latest/format.html#parameters, where the separation between types is an "or" and between the list of types and info like "optional" is a comma.

Reply:
Also, yes, this consistency does help with programmatic parsing of docstrings.

Reply:
Since the docstring type annotation mimics the real type annotation, I've also seen a lot of new docstrings swap the "or" for "|"
```
index : int | str | None, optional
```

Reply:
I think this is actually the most readable.

Reply:
I don't see the text state that more than 2 options need multiple "or".  Am I reading over it?

Reply:
p.s. I sort-of like the `|` option, but feel it takes away a bit from the purpose of a docstring, which is to give a description in english, aimed at humans... Which is why I prefer "A, B or C".

Reply:
> I don't see the text state that more than 2 options need multiple "or". Am I reading over it?

Hm. I guess it's not explicit. I glanced at Pandas and they use commas in their lists. IMO it slows down my parsing of types versus comments like optional, keyword-only, etc. with "or" or "|" I see it as a block. With "," they are mixed together.

```
param : this or that or something, optional
```

vs

```
param : this, that, or something, optional
```

The comma works as a delimiter, making the second option muddled compared to the first.

Reply:
For me it is less obvious, although I would put "optional" in parentheses (as you had it) or, more likely, use a semi-colon.  But repeating "or" is not really grammatical... Note that it is slightly better if you do not use a comma before "or" (which is ApJ-style, but not that commonly done elsewhere).

Reply:
I've reverted this.